sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/DirectoryAsyncClient.java

```diff
@@ -150,28 +152,39 @@ public DirectoryAsyncClient getSubDirectoryClient(String subDirectoryName) {
      *
      * <p>Create the directory</p>
      *
-     * {@codesnippet com.azure.storage.file.directoryAsyncClient.createWithResponse#Map}
+     * {@codesnippet com.azure.storage.file.directoryAsyncClient.createWithResponse#filesmbproperties-string-map}
      *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/create-directory">Azure Docs</a>.</p>
      *
+     * @param smbProperties The SMB properties of the directory.
+     * @param filePermission The file permission of the directory.
      * @param metadata Optional metadata to associate with the directory
      * @return A response containing the directory info and the status of creating the directory.
      * @throws StorageException If the directory has already existed, the parent directory does not exist or directory name is an invalid resource name.
      */
-    public Mono<Response<DirectoryInfo>> createWithResponse(Map<String, String> metadata) {
-        return withContext(context -> createWithResponse(metadata, context));
-    }
-
-    Mono<Response<DirectoryInfo>> createWithResponse(Map<String, String> metadata, Context context) {
-        // TODO (alzimmer): These properties are dummy defaults to allow the new service version to be used. Remove these and use correct defaults when known (https://github.com/Azure/azure-sdk-for-java/issues/5039)
-        String fileAttributes = "None";
-        String filePermission = "inherit";
-        String fileCreationTime = "now";
-        String fileLastWriteTime = "now";
-        return postProcessResponse(azureFileStorageClient.directorys().
-            createWithRestResponseAsync(shareName, directoryPath, fileAttributes, fileCreationTime, fileLastWriteTime,
-                null, metadata, filePermission, null, context))
+    public Mono<Response<DirectoryInfo>> createWithResponse(FileSmbProperties smbProperties, String filePermission,
```

Comment:
The way this is exposed is different in directories and files. I put the filePermission in the file properties which may or may not make sense since it is never returned from the service in these calls. I could take filePermission out of the file fileproperties and make it look closer to the way directories expose it

Reply:
I think it's probably best to be consistent between the two.

Reply:
Currently working on extracting filePermission